src/Umbraco.Examine/ContentIndexPopulator.cs

```diff
@@ -70,31 +70,65 @@ protected override void PopulateIndexes(IReadOnlyList<IIndex> indexes)
             {
                 contentParentId = _parentId.Value;
             }
+
+            if (_publishedValuesOnly)
+            {
+                IndexPublishedContent(contentParentId, pageIndex, pageSize, indexes);
+            }
+            else
+            {
+                IndexAllContent(contentParentId, pageIndex, pageSize, indexes);
+            }
+        }
+
+        protected void IndexAllContent(int contentParentId, int pageIndex, int pageSize, IReadOnlyList<IIndex> indexes)
+        {
             IContent[] content;
 
             do
             {
-                if (!_publishedValuesOnly)
-                {
-                    content = _contentService.GetPagedDescendants(contentParentId, pageIndex, pageSize, out _).ToArray();
-                }
-                else
+                content = _contentService.GetPagedDescendants(contentParentId, pageIndex, pageSize, out _).ToArray();
```

Comment:
ToList is faster than ToArray

Reply:
I think an array is used here because the  GetValueSets method of  ValueSetBuilder needs an array. 
Don't know if using a list and then calling ToArray is faster ?

Reply:
Ah i see, no worries then. And nope, calling ToList() then ToArray() isn't faster, it will be slow and also allocate another unnecessary object.